games-simulation/openrct2/openrct2-0.2.4.ebuild

```diff
@@ -0,0 +1,182 @@
+		!libressl? ( dev-libs/openssl:0= )
+	)
+"
+
+RDEPEND="
+	${DEPEND_RDEPEND}
+	dedicated? (
+		acct-group/openrct2
+		acct-user/openrct2
+	)
+"
+
+DEPEND="
+	${DEPEND_RDEPEND}
+	app-arch/unzip
+	test? ( dev-cpp/gtest )
+"
+
+BDEPEND="virtual/pkgconfig"
+
+RESTRICT="!test? ( test )"
+
+S="${WORKDIR}/${MY_PN}-${PV}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.2.4-disable-desktop-and-icon-files-without-gui.patch"
+	"${FILESDIR}/${PN}-0.2.4-disable-optional-features.patch"
+	"${FILESDIR}/${PN}-0.2.4-include-additional-paths.patch"
+)
+
+src_unpack() {
+	unpack "${P}".tar.gz
+
+	mkdir -p "${S}"/data/title || die
+	cd "${S}"/data/title || die
+	unpack "${PN}-${MY_PN_TS}-${MY_PV_TS}".zip
+
+	mkdir -p "${S}"/data/object || die
+	cd "${S}"/data/object || die
+	unpack "${PN}-${MY_PN_OBJ}-${MY_PV_OBJ}".zip
+
+	if use test; then
+		mkdir -p "${S}"/testdata/replays || die
+		cd "${S}"/testdata/replays || die
+		unpack "${PN}-${MY_PN_RPL}-${MY_PV_RPL}".zip
+	fi
+}
+
+src_prepare() {
+	cmake-utils_src_prepare
+
+	# Don't treat warnings as errors.
+	sed -e 's/-Werror//' -i CMakeLists.txt || die
+}
+
+src_configure() {
+	# Note: There is currently no support for Disord-RPC and Google Benchmark,
+	# as both packages do not exist in Gentoo, so support for them has been disabled.
+	local mycmakeargs=(
+		-DDISABLE_DISCORD_RPC=ON
+		-DDISABLE_GOOGLE_BENCHMARK=ON
+		-DDISABLE_GUI=$(usex dedicated)
+		-DDISABLE_HTTP_TWITCH=$(usex !twitch)
+		-DDISABLE_NETWORK=$(usex !multiplayer)
+		$(usex !dedicated "-DDISABLE_OPENGL=$(usex !opengl)" "")
+		-DDISABLE_TTF=$(usex !truetype)
+		-DDOWNLOAD_OBJECTS=OFF
+		-DDOWNLOAD_REPLAYS=OFF
+		-DDOWNLOAD_TITLE_SEQUENCES=OFF
+		-DENABLE_LIGHTFX=$(usex lightfx)
+		-DPORTABLE=OFF
+		-DSTATIC=OFF
+		$(usex test "-DSYSTEM_GTEST=ON" "")
+		-DWITH_TESTS=$(usex test)
+		-DUSE_MMAP=ON
+	)
+
+	cmake-utils_src_configure
+}
+
+src_test() {
+	# Since the tests need the OpenRCT2 data,
+	# we need to copy them into the build directory,
+	# otherwise some tests will fail, as they don't find the OpenRCT2 data.
+	# It is currently not possible to override the path.
+	# See: https://github.com/OpenRCT2/OpenRCT2/issues/6473
+	cp -r "${S}"/data "${BUILD_DIR}" || die
+
+	cmake-utils_src_test
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	if use dedicated; then
+		newinitd "${FILESDIR}"/openrct2.initd openrct2
+		newconfd "${FILESDIR}"/openrct2.confd openrct2
+	fi
+}
+
+pkg_postinst() {
+	if [[ -z ${REPLACING_VERSIONS} ]]; then
+		einfo "OpenRCT2 needs the original RollerCoaster Tycoon 2 data files."
+		einfo ""
+		einfo "You need the folders 'Data', 'ObjData', 'Scenarios' and 'Tracks'"
+		einfo "from the original release and copy them into '/usr/share/openrct2/rct2'."
+		einfo ""
+		einfo "For additional content, you can install the RollerCoaster Tycoon 1 data files."
+		einfo ""
+		einfo "You need the folders 'Data', 'Scenarios' and 'Tracks'"
+		einfo "from the original release and copy them into '/usr/share/openrct2/rct1'."
```

Comment:
Thanks, added. Would it be valid also to use `${ROOT%/}/path/to/foo`? I am not a fan of removing slash.

Reply:
Well, the question should be now answered.. but the example in devmanual is also wrong.
'repoman' does not complain at all, but 'pkgcheck scan' does.

Since Gentoo CI on Github uses `pkgcheck`, the correct answer is `${ROOT}/path/to/foo`.
Pushed.

Reply:
> Well, the question should be now answered.. but the example in devmanual is also wrong.
> `repoman -dx full` does not complain at all, but `pkgcheck scan` does.
> 
> It's sometime hard to make it correct, when the examples in devmanual are wrong :(
> 

Wrong how? Can you explain it a bit more please?

Reply:
> Thanks, added. Would it be valid also to use `${ROOT%/}/path/to/foo`? I am not a fan of removing slash.

There is relevant information about this in the change log for EAPI=7. Can you quote me that?

Reply:
> Wrong how? Can you explain it a bit more please?

The example, which is shown in your link, is at least not valid for EAPI=7.
As a developer, I would expect, that examples are probably updated for EAPI=7, so for now, I see, that I cannot rely on devmanual.

Example shows `${ROOT}etc/newconfig`. If you do set this, `pkgcheck` and Github CI will complain, but not `repoman`. I guess, more reasons to use only `pkgcheck`.

> There is relevant information about this in the change log for EAPI=7. Can you quote me that?

It's in chapter `11.1` and table `11.7`. With EAPI=7, `ROOT` (and a bunch of other variables) does not have any trailing slash anymore, so I need to add/specify it. My first example with stripping slash does not make anyway sense, since there is nothing to strip.

Reply:
Fixed, thanks. https://gitweb.gentoo.org/proj/devmanual.git/commit/?id=22e517a342ebf782fdb39afb4c7da23f6a9b4413

Reply:
Thanks :-)